jenkins/ci.suse.de/openstack-mkcloud.yaml

```diff
@@ -418,251 +418,19 @@
+          ${automationrepo}/scripts/jenkins/update_automation
 
-          fi
-
-          cp ${automationrepo}/scripts/jenkins/log-parser/openstack-mkcloud-rules.txt \
-              $WORKSPACE/log-parser-plugin-rules.txt
-
-          echo "########################################################################"
-          env
-          echo "########################################################################"
-
-          MKCLOUDTARGET=$mkcloudtarget
-          [ $UPDATEBEFOREINSTALL == "true" ] && MKCLOUDTARGET='cleanup prepare setupadmin addupdaterepo instcrowbar setupcompute instcompute proposal testsetup'
-
-          [ $(uname -m) = s390x ] && WITHCROWBARREGISTER=true
-          if [ $WITHCROWBARREGISTER == "true" ] ; then
-            export nodenumberlonelynode=1
-            [ $(uname -m) = s390x ] && {
-                export nodenumberlonelynode=$nodenumber
-                export nodenumber=0
-                export controller_node_memory=8388608
-                export compute_node_memory=6291456
-            }
-            MKCLOUDTARGET+=" setuplonelynodes crowbar_register"
-          fi
-
-          [ $(uname -m) = aarch64 ] && {
-              export vcpus=16
-          }
-
-          pushd ~/pool/
-              # free up a pool if more than one is reserved
-              # policy allows only one reservation per host at the moment
-              ls -1 | grep -v "^[0-9]\+$" | sort -R | head -n -1 | while read p ; do mv $p ${p%%.*} ; done
-          popd
-
-          starttime=`date +%s`
-
-          # CHECKPROC in env tells 'allocpool' the process name to compare
-          export CHECKPROC=mkcloud
-          mkcloudwrapper="${automationrepo}/scripts/mkcloudhost/allocpool ${automationrepo}/scripts/mkcloudhost/mkcloude"
-
-          export clouddescription="Jenkins Build: $BUILD_NUMBER / $JOB_NAME"
-
-          $custom_settings
-
-          perl -e "alarm 6*60*60 ; exec '${mkcloudwrapper} bash ${automationrepo}/scripts/mkcloud setuphost $(echo -n $MKCLOUDTARGET) ' ; print qq{$!\n} ; exit 127 " | tee $artifacts_dir/mkcloud_short_stdout.log
-          ret=${PIPESTATUS[0]}
-          if [[ $ret != 0 ]] ; then
-              if [[ $github_pr_sha ]] ; then
-                mkcloudgating_trap
-              else
-                $jtsync --ci suse --job $JOB_NAME 1
-              fi
-              echo "mkcloud ret=$ret"
-              exit $ret # check return code before tee
-          fi
-
-          # report mkcloud-gating status or jenkins trello status
-          if [[ $github_pr_sha ]] ; then
-            $ghpr --action set-status --debugratelimit $ghpr_paras --status "success" --targeturl $BUILD_URL --message "PR gating succeeded"
-            trap "-" ERR
-          else
-            $jtsync --ci suse --job $JOB_NAME 0
-          fi
+          # run the builder script in the automation repo
+          ${automationrepo}/scripts/jenkins/mkcloud/openstack-mkcloud.builder.sh
```

Comment:
Could we prepend an `exec` here? I sort of remember there being issues with canceling jobs, where only the process spawned by jenkins got a signal, but not its children, which caused inconsistencies with slot allocation vs jenkins executors.